### PR TITLE
Use exact capitalization from field names overridden in config

### LINKED ISSUE #1447: should be able to override go field capitalization
### What happened?

I've inherited code that currently uses gqlgen v0.7.1.  I'm trying to upgrade to 0.13.0, to take advantage of new features.

There's a change to generated field capitalization which ripples through this inherited code base.  I'm hoping to override gqlgen's current preferred capitalization.  I haven't found any way to do that (short of defining the models manually).

For example, if I define a gql type...

```
type Something {
  somethingDDL: Int!
  somethingGB: Int!
}
```

the generated go type is...

```
type Something struct {
  SomethingDdl int // I want "SomethingDDL" here
  SomethingGb int // I want "SomethingGB" here
}
```

I've tried to set the name using `@goField(name: "SomethingDDL")` directives.  And I've tried to set the field name in the gqlgen.yml file.  Either I haven't done it just right, or the capitalization of generated names cannot be overridden.

### What did you expect?

I'd like to better control the capitalization of generated field names.  If I could, I'd make the generated code capitalize the first letter of the GQL field name, and leave the other letters unchanged.

It would be a nice feature to have, because gqlgen is opinionated when it comes to capitalizing field names.  It insists that "Id" (not an acronym) be spelled "ID", while "GB" (an acronym) be spelled "Gb".  And this has changed sometime between version 0.7 and 0.13.

### Minimal graphql.schema and models to reproduce

Sorry I haven't put together a minimal schema.

### versions
 - `gqlgen version`? 
 0.13.0

 - `go version`?
1.15.2

 - dep or go modules?
modules

### PR DESCRIPTION
**Pull request by ianling**
Closes #1447 

Behavior prior to this PR: given the following schema and gqlgen config:

Schema:
```graphql
type RenameFieldTest {
    badName: String!
    otherField: String!
}
```

gqlgen.yml:
```yml
models:
  RenameFieldTest:
    fields:
      badName:
        fieldName: GOODnaME
```

gqlgen generates the following Go struct:

```go
type RenameFieldTest struct {
	GOODnaMe   string `json:"badName"`  // <----- field name should be GOODnaME, but e is lowercase
	OtherField string `json:"otherField"`
}
```

Behavior after this PR (modelgen strictly adheres to field names overridden in the config): 

```go
type RenameFieldTest struct {
	GOODnaME   string `json:"badName"` // <----- the 'e' is now correctly capitalized
	OtherField string `json:"otherField"`
```

Field names that are not overridden in the config will be exactly the same as before this PR, so users will only see different behavior if they had an overridden name that happened to fall into one of these funny capitalization edge cases.

I have:
 - [x] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [x] ~~Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))~~ (not necessary)

### CODE REVIEW

**Comment by StevenACoffman** [comment]
Thanks! Nice catch!

File: plugin/modelgen/models.gotpl [hunk 1]
```diff
@@ -29,7 +29,7 @@
 			{{- with .Description }}
 				{{.|prefixLines "// "}}
 			{{- end}}
-			{{ $field.Name|go }} {{$field.Type | ref}} `{{$field.Tag}}`
```
Comment (carldunham) [inline plugin/modelgen/models.gotpl:32]: So is this because of a bug in the `go` filter?

Comment (ianling) [reply plugin/modelgen/models.gotpl:32]: Nope, the `go` filter was just being erroneously applied in situations where the field name and its capitalization were being explicitly overridden in the gqlgen config. This was a problem because that filter changes the capitalization of field names that are already valid in Go in a slightly opinionated way, which was unwanted behavior in this case.

Comment (carldunham) [reply plugin/modelgen/models.gotpl:32]: Got it. Cool.

Comment (YogeshLele) [reply plugin/modelgen/models.gotpl:32]: @ianling Is there a way (in config or otherwise) to disable this?

Comment (ianling) [reply plugin/modelgen/models.gotpl:32]: @YogeshLele disable what exactly? If you want to specify a particular capitalization for a field in your schema, this PR made it so you can do so without gqlgen overriding it.

**Comment by MohitVachhani** [comment]
Will this work for the cases of enums ?
